aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/BaseHandlerStd.java

```diff
@@ -221,12 +222,25 @@ protected boolean isGlobalClusterMember(final ResourceModel model) {
         return StringUtils.isNotBlank(model.getGlobalClusterIdentifier());
     }
 
+    protected boolean isDBClusterAvailable(final DBCluster dbCluster) {
+        return DBClusterStatus.Available.equalsString(dbCluster.status());
+    }
+
+    protected boolean isNoPendingChanges(final DBCluster dbCluster) {
+        final ClusterPendingModifiedValues modifiedValues = dbCluster.pendingModifiedValues();
```

Comment:
Maintenance actions are applied during maintenance. I believe these changes are applied in a few cases: 
* After [ApplyPendingMaintenanceAction] API call (https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_ApplyPendingMaintenanceAction.html)
* During Maintenance Window
* After ModifyDBCluster call with --applyImmediately

Sometimes there is a short delay before pending maintenance action is picked by RDS after a modify call. In that case the stabilization logic could run forever (or rather up to 36 hours or until next maintenance). Can you confirm that this is not the case?